community/cypher/frontend-3.2/src/test/scala/org/neo4j/cypher/internal/frontend/v3_2/prettifier/PrettyTest.scala

```diff
@@ -0,0 +1,200 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.frontend.v3_2.prettifier
+
+import org.neo4j.cypher.internal.frontend.v3_2.parser.CypherParser
+import org.neo4j.cypher.internal.frontend.v3_2.test_helpers.CypherFunSuite
+
+class PrettyTest extends CypherFunSuite {
+
+  test("should upcase keywords") {
+    actual("create (n)") should equal(expected("CREATE (n)"))
+  }
+
+  test("should not break INDEX ON") {
+    actual("create index on :Person(name)") should equal(expected("CREATE INDEX ON :Person(name)"))
+  }
+
+  test("should not break on ASC") {
+    actual("RETURN n order by n.name asc") should equal(expected("RETURN n ORDER BY n.name ASC"))
+  }
+
+  test("should not break CREATE in FOREACH") {
+    actual("match p=(n) foreach(x in p | create (x)--())") should equal(expected("MATCH p = (n)%nFOREACH (x IN p |%n  CREATE (x)--()%n)"))
+  }
+
+  test("should not break CREATE in complex FOREACH") {
+    actual("match p=(n) foreach(x in p | create (x)--() set x.foo = 'bar') return distinct p;") should equal(
+      expected(
+        "MATCH p = (n)%n" +
+          "FOREACH (x IN p |%n" +
+          "  CREATE (x)--()%n" +
+          "  SET x.foo = 'bar'%n" +
+          ")%n" +
+          "RETURN DISTINCT p")
+    )
+  }
+
+  test("should not break STARTS WITH ") {
+    actual("return 'apartment' starts with 'apa' as x") should equal(
+      expected("RETURN 'apartment' STARTS WITH 'apa' AS x")
+    )
+  }
+
+  test("should not break ENDS WITH ") {
+    actual("return 'apartment' ends with 'apa' as x") should equal(
+      expected("RETURN 'apartment' ENDS WITH 'apa' AS x")
+    )
+  }
+
+  test("should not break CONSTRAINT ON") {
+    actual("create constraint on (person:Person) assert person.age is unique") should equal(
+      expected("CREATE CONSTRAINT ON (person:Person) ASSERT person.age IS UNIQUE")
+    )
+  }
+
+  test("should break ON CREATE") {
+    actual("merge (n) on create set n.age=32") should equal(expected("MERGE (n)%n  ON CREATE SET n.age = 32"))
+  }
+
+  test("should correctly handle parenthesis in MATCH") {
+    actual("match (a)-->(b) return b") should equal(expected("MATCH (a)-->(b)%nRETURN b"))
+  }
+
+  test("should upcase multiple keywords") {
+    actual("match (n) where n.name='B' return n") should equal(expected("MATCH (n)%n  WHERE n.name = 'B'%nRETURN n"))
+  }
+
+  test("should upcase multiple keywords 2") {
+    actual("match (a) where a.name='A' return a.age as SomethingTotallyDifferent") should equal(
+      expected("MATCH (a)%n  WHERE a.name = 'A'%nRETURN a.age AS SomethingTotallyDifferent")
+    )
+  }
+
+  test("should not break CREATE UNIQUE") {
+    actual("start me=node(3) match p1=(me)-[*2]-(friendOfFriend) " +
```

Comment:
Do we support `start n=node(#)` syntax in any recent version? I thought this was removed already in 2.1?